File: Kiosk/kiosk_order/src/order_listener.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import String

def callback(data):
    rospy.loginfo("주문 수신: %s", data.data)

def listener():
    rospy.init_node('order_listener', anonymous=True)
    rospy.Subscriber('/order_topic', String, callback)
    rospy.spin()

if __name__ == '__main__':
    listener()

File: Kiosk/kiosk_order/src/order_listener.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void orderCallback(const std_msgs::String::ConstPtr& msg)
{
    ROS_INFO("주문 수신: %s", msg->data.c_str());
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "order_listener");
    ros::NodeHandle n;

    ros::Subscriber sub = n.subscribe("/order_topic", 1000, orderCallback);

    ros::spin();
    return 0;
}
